rust/pact_matching/src/lib.rs

```diff
@@ -2103,22 +2105,19 @@ pub async fn generate_request(request: &HttpRequest, mode: &GeneratorTestMode, c
           if let Some(parameter) = parameters.get_mut(param) {
             let mut generated = parameter.clone();
             for (index, val) in parameter.iter().enumerate() {
-              if let Ok(v) = generator.generate_value(val, context, &DefaultVariantMatcher.boxed()) {
-                generated[index] = v;
+              let value = val.clone().unwrap_or_default();
+              if let Ok(v) = generator.generate_value(&value, context, &DefaultVariantMatcher.boxed()) {
+                generated[index] = Some(v);
```

Comment:
As above, just double checking whether the loss of distinction between `Some("")` and `None` is intentional.

Reply:
Generator code, value doesn't matter.